### PR TITLE
logger: Add SetAutoJSONFormat to select JSON or console output

### PR DESCRIPTION
**Pull request by jiping-s**

### CODE REVIEW

File: logger/logger.go [hunk 1]
```diff
@@ -113,6 +115,21 @@ func SetAutoFormat() {
 	}
 }
 
+// SetAutoJSONFormat uses the environment variable `LOG_COLOR` and terminal detection to select console or JSON output format
+func SetAutoJSONFormat() {
+	colorYN := strings.ToLower(os.Getenv("LOG_COLOR"))
```
Comment (jstaffans) [inline logger/logger.go:120]: What about following https://no-color.org/ instead and having a `NO_COLOR` env var? 

* Default is color (no env var needed)
* When deployed, set a `NO_COLOR` env var which also forces JSON format

**Review by jstaffans — APPROVED** [review]

File: logger/logger.go [hunk 2]
```diff
@@ -113,6 +115,21 @@ func SetAutoFormat() {
 	}
 }
 
+// SetAutoJSONFormat uses the environment variable `LOG_COLOR` and terminal detection to select console or JSON output format
+func SetAutoJSONFormat() {
```
Comment (jstaffans) [inline logger/logger.go:119]: Is this function actually called from anywhere?

Comment (jiping-s) [reply logger/logger.go:119]: Not yet, it's meant to replace SetJSONFormat so we can still see colored logs during development

File: logger/logger.go (hunk 1 above)
Comment (jiping-s) [reply logger/logger.go:120]: The default is done by detecting whether stderr is a terminal or not, while the env var can be used to override the behavior (in gitlab CI for example, where color codes are translated to HTML). Normally we don't need to set anything whether on production or for development.

Comment (jstaffans) [reply logger/logger.go:120]: OK, sounds fine to me.

**Pull request merged**